packages/frontend/src/react-app-env.d.ts

```diff
@@ -5,9 +5,11 @@ import { resources } from "./i18n/resources";
 import { CARROT_KPI_FRONTEND_I18N_NAMESPACE } from "./constants";
 
 declare global {
-    const __PROD__: boolean;
-    const __LIBRARY_MODE__: boolean;
-    const __STAGING_MODE__: boolean;
+    const __BUILDING_MODE__:
+        | "production"
+        | "staging"
+        | "library"
```

Comment:
We're adding `library` here but in the `ALLOWED_BUILDING_MODES` we don't have that, is this expected?

Reply:
Yes because the `craco` configuration cannot be used to build the library mode version of the dapp. That's why it's only available (and hardcoded) in the rollup build config.